bundles/org.eclipse.ui.workbench.texteditor/src/org/eclipse/ui/internal/findandreplace/IFindReplaceLogic.java

```diff
@@ -174,4 +174,10 @@ public interface IFindReplaceLogic {
 	 */
 	public boolean isWholeWordSearchAvailable(String findString);
 
+	/**
+	 * Initializes the anchor used as starting point for incremental searching.
+	 * Subsequent incremental searches will start from the current selection.
```

Comment:
Maybe you could state what "the current selection" is. Precisely, this is the current selection in the target, isn't it? I would also appreciate a short explanation about how range selections instead of single positions are or should be treated. My current understanding is: forward search operations uses the beginning of the selection, backward search operations use the end of the selection, is that correct?

This allows us in the future to better decide whether some behavior is correct/intended or not and how tests according to the spec need to be written properly.

Reply:
https://github.com/eclipse-platform/eclipse.platform.ui/pull/2104#discussion_r1684201267

improved specification!